File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Java/Ejem09MenusContextualesydeAplicacion/Ejem09MenusContextualesydeAplicacion/app/src/main/java/com/example/ejem09menuscontextualesydeaplicacion/MainActivity.java
package com.example.ejem09menuscontextualesydeaplicacion;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {
 
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Registramos el menu contextual en el layout principal
        registerForContextMenu(findViewById(R.id.ly_principal));
    }
    //Se encarga de visualizar el menu de aplicacion y de construir el menu a partir del xml que
    //le indiquemos
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.mi_menu, menu);
        return true;
    }

    //Metodo para gestionar la opción seleccionada
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        /* No funciona con switch
        if(item.getItemId()== R.id.menu_opcion1){
           Log.d("depurando","Opcion 1 pulsada");
        }*/
        Snackbar.make(findViewById(R.id.ly_principal), item.getTitle().toString(),Snackbar.LENGTH_LONG).show();
        return super.onOptionsItemSelected(item);
    }

    // IMPLEMENTAR MENU CONTEXTUAL
    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
        getMenuInflater().inflate(R.menu.mi_menu, menu);
    }
    //Recuperamos el metodo de las opciones seleccionadas del otro menu creado
    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {
        return onOptionsItemSelected(item);
    }
}

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejercicios/Kotlin/Ejercicio010_DialogoConfirmacion/app/src/main/java/com/example/ejercicio010_dialogoconfirmacion/MainActivity.kt
package com.example.ejercicio010_dialogoconfirmacion

import android.app.AlertDialog
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuItem
import android.view.View
import com.example.ejercicio010_dialogoconfirmacion.databinding.ActivityMainBinding
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        registerForContextMenu(binding.lLayout)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return super.onOptionsItemSelected(item)
    }

    private fun confirmar(color:String) {
        var colorSeleccionado: Int =0
        when(color){
            "Rojo" -> colorSeleccionado= Color.RED;
            "Verde" -> colorSeleccionado=Color.GREEN;
            "Amarillo" -> colorSeleccionado=Color.YELLOW;
            "Blanco" -> colorSeleccionado=Color.WHITE;
        }
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Cambio de color")
        builder.setMessage("Vas a cambiar el color de fondo a "+color+", estas seguro?")
        builder.setPositiveButton("Si"){dialog, which ->
            binding.lLayout.setBackgroundColor(colorSeleccionado)
            dialog.cancel()
        }
        builder.setNegativeButton("no"){dialog, which->
            Snackbar.make(binding.root, "Pues vale", Snackbar.LENGTH_SHORT).show()
            dialog.cancel()
        }
        builder.show()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onCreateContextMenu(
        menu: ContextMenu?,
        v: View?,
        menuInfo: ContextMenu.ContextMenuInfo?
    ) {
        menuInflater.inflate(R.menu.menu, menu)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        confirmar(item.title.toString())
        return true
    }
}